master/buildbot/reporters/pushjet.py

```diff
@@ -0,0 +1,116 @@
+                    mode=("failing", "passing", "warnings"),
+                    tags=None, builders=None,
+                    buildSetSummary=False, messageFormatter=None,
+                    subject="Buildbot %(result)s in %(title)s on %(builder)s",
+                    name=None, schedulers=None, branches=None,
+                    levels=None, base_url='https://api.pushjet.io',
+                    watchedWorkers=None, messageFormatterMissingWorker=None):
+
+        super(PushjetNotifier, self).checkConfig(mode, tags, builders,
+                                                 buildSetSummary, messageFormatter,
+                                                 subject, False, False,
+                                                 name, schedulers,
+                                                 branches, watchedWorkers)
+
+        httpclientservice.HTTPClientService.checkAvailable(self.__class__.__name__)
+
+    @defer.inlineCallbacks
+    def reconfigService(self, secret,
+                        mode=("failing", "passing", "warnings"),
+                        tags=None, builders=None,
+                        buildSetSummary=False, messageFormatter=None,
+                        subject="Buildbot %(result)s in %(title)s on %(builder)s",
+                        name=None, schedulers=None, branches=None,
+                        levels=None, base_url='https://api.pushjet.io',
+                        watchedWorkers=None, messageFormatterMissingWorker=None):
+
+        if messageFormatter is None:
+            messageFormatter = DefaultMessageFormatter(template_type='html',
+                template_filename='default_notification.txt')
+        if messageFormatterMissingWorker is None:
+            messageFormatterMissingWorker = MessageFormatterMissingWorker(
+                template_filename='missing_notification.txt')
+        super(PushjetNotifier, self).reconfigService(mode, tags, builders,
+                                                     buildSetSummary, messageFormatter,
+                                                     subject, False, False,
+                                                     name, schedulers, branches,
+                                                     watchedWorkers, messageFormatterMissingWorker)
+        self.secret = secret
+        if levels is None:
+            self.levels = {}
+        else:
+            self.levels = levels
+        self._http = yield httpclientservice.HTTPClientService.getService(
+            self.master, base_url)
+
+    def sendMessage(self, body, subject=None, type=None, builderName=None,
+                    results=None, builds=None, users=None, patches=None,
+                    logs=None, worker=None):
+
+        if worker is not None and worker not in self.watchedWorkers:
+            return
+
+        msg = {'message': body}
+        try:
+            msg['level'] = self.levels[LEVELS[results] if worker is None else 'worker_missing']
```

Comment:
I would prefer this is implemented with self.levels.get(xx, default) and avoid the untested try except path.